data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/IngressProducerReconcilableStore.java

```diff
@@ -89,9 +89,27 @@ public IngressProducer resolve(String host, String path) {
     // That means, we support these modes:
     // - Request coming to "/path" --> path is used for matching
     // - Request coming to "/" --> hostname is used for matching
-    if (pathMapper.get(path) != null) {
-      return pathMapper.get(path);
+    final var p = pathMapper.get(path);
+    if (p != null) {
+      return p;
     }
+
+    // Host based routing is not as simple as path based routing, since the host header
+    // is variable.
+    // For example an external name service might be hit in multiple ways producing
+    // different host headers:
+    // - my-kafka-channel-kn-channel.default
+    // - my-kafka-channel-kn-channel.default:80
+    // - my-kafka-channel-kn-channel.default.svc
+    // - my-kafka-channel-kn-channel.default.svc:80
+    // - my-kafka-channel-kn-channel.default.svc.cluster.local [1]
+    // - my-kafka-channel-kn-channel.default.svc.cluster.local:80
+    //
+    // Current implementations expects only the long form that contains the cluster
+    // domain without the port [1], however, this is clearly a non-robust implementation,
```

Comment:
PErhaps worth to create an issue for a different impl... but the names should, on the cluster, resolve to same, regardless if long form or short

Reply:
right, that's what I mean with this comment, but for routing, a simple hash map doesn't work!